packages/bsky/src/api/app/bsky/unspecced/getPostThreadV2.ts

```diff
@@ -0,0 +1,154 @@
+import { isThreadItemNotFound } from '@atproto/api/dist/client/types/app/bsky/unspecced/getPostThreadV2'
+import { InvalidRequestError } from '@atproto/xrpc-server'
+import { ServerConfig } from '../../../../config'
+import { AppContext } from '../../../../context'
+import { Code, DataPlaneClient, isDataplaneError } from '../../../../data-plane'
+import { HydrateCtx, Hydrator } from '../../../../hydration/hydrator'
+import { Server } from '../../../../lexicon'
+import {
+  OutputSchema,
+  QueryParams,
+} from '../../../../lexicon/types/app/bsky/unspecced/getPostThreadV2'
+import {
+  HydrationFnInput,
+  PresentationFnInput,
+  SkeletonFnInput,
+  createPipeline,
+  noRules,
+} from '../../../../pipeline'
+import { postUriToThreadgateUri } from '../../../../util/uris'
+import { Views } from '../../../../views'
+import { ATPROTO_REPO_REV, resHeaders } from '../../../util'
+
+export default function (server: Server, ctx: AppContext) {
+  const getPostThread = createPipeline(
+    skeleton,
+    hydration,
+    noRules, // handled in presentation: 3p block-violating replies are turned to #blockedPost, viewer blocks turned to #notFoundPost.
+    presentation,
+  )
+  server.app.bsky.unspecced.getPostThreadV2({
+    auth: ctx.authVerifier.optionalStandardOrRole,
+    handler: async ({ params, auth, req, res }) => {
+      const { viewer, includeTakedowns, include3pBlocks } =
+        ctx.authVerifier.parseCreds(auth)
+      const labelers = ctx.reqLabelers(req)
+      const hydrateCtx = await ctx.hydrator.createContext({
+        labelers,
+        viewer,
+        includeTakedowns,
+        include3pBlocks,
+      })
+
+      let result: OutputSchema
+      try {
+        result = await getPostThread({ ...params, hydrateCtx }, ctx)
+      } catch (err) {
+        const repoRev = await ctx.hydrator.actor.getRepoRevSafe(viewer)
+        if (repoRev) {
+          res.setHeader(ATPROTO_REPO_REV, repoRev)
+        }
+        throw err
+      }
+
+      const repoRev = await ctx.hydrator.actor.getRepoRevSafe(viewer)
```

Comment:
This is used for read-after-write.  No harm including it, but if the client and pds aren't going to use it, we may be able to start by omitting it.

Reply:
I'd recommend not doing it if we can get away with it, since we are trying to move away from app logic in the pds.  The endpoint is also still unspecced, potentially prone to change.